src/js/components/deployments/createdeployment.js

```diff
@@ -99,9 +100,20 @@ export class CreateDialog extends React.Component {
   onScheduleSubmit(settings) {
     const self = this;
     const { hasNewRetryDefault } = self.state;
-    const { advanceOnboarding, createDeployment, globalSettings, isOnboardingComplete, onScheduleSubmit, saveGlobalSettings } = self.props;
+    const {
+      advanceOnboarding,
+      createDeployment,
+      globalSettings,
+      isEnterprise,
+      isHosted,
+      isOnboardingComplete,
+      onScheduleSubmit,
+      plan,
+      saveGlobalSettings
+    } = self.props;
     const { deploymentDeviceIds, device, filterId, group, phases, release, retries } = settings;
     const startTime = phases?.length ? phases[0].start_ts || new Date() : new Date();
+    const retrySetting = isEnterprise || (isHosted && plan !== PLANS.os.value) ? { retries } : {};
```

Comment:
I had this vision about an interface like that:
```js
class Plan { // or of course function ;-)
/* ... */
}
plan=new Plan(PLANS.os.value); // "pseudo" code, just to note that we have some plan
/* ... */
if (plan.canAccess(FEATURES.RETRIED_DEPLOYMENTS) { // canAccess does all the magic to check if the plan includes a feature
 /* retrySetting = ... */
}
```

Reply:
we have something similar [here](https://github.com/mendersoftware/gui/blob/master/src/js/selectors/index.js#L52-L85) - but I would need some focused time on that to move some more decisions in there...

Reply:
ah, yes I remember. right! thank you for taking time to answer.